mleap-core/src/main/scala/ml/combust/mleap/core/types/Casting.scala

```diff
@@ -245,32 +253,88 @@ object Casting {
-                  (v: Any) => v.asInstanceOf[Tensor[_]].mapValues(cc)
+                  (v: Any) => {
+                    v match {
+                      case _: SparkVector => sparkVectorToMLeapTensor(v.asInstanceOf[SparkVector]).mapValues(cc)
+                      case _ => v.asInstanceOf[Tensor[_]].mapValues(cc)
+                    }
+                  }
                 case BasicType.Long =>
                   val cc = c.asInstanceOf[(Any) => Long]
-                  (v: Any) => v.asInstanceOf[Tensor[_]].mapValues(cc)
+                  (v: Any) => {
+                    v match {
+                      case _: SparkVector => sparkVectorToMLeapTensor(v.asInstanceOf[SparkVector]).mapValues(cc)
+                      case _ => v.asInstanceOf[Tensor[_]].mapValues(cc)
+                    }
+                  }
                 case BasicType.Float =>
                   val cc = c.asInstanceOf[(Any) => Float]
-                  (v: Any) => v.asInstanceOf[Tensor[_]].mapValues(cc)
+                  (v: Any) => {
+                    v match {
+                      case _: SparkVector => sparkVectorToMLeapTensor(v.asInstanceOf[SparkVector]).mapValues(cc)
+                      case _ => v.asInstanceOf[Tensor[_]].mapValues(cc)
+                    }
+                  }
                 case BasicType.Double =>
                   val cc = c.asInstanceOf[(Any) => Double]
-                  (v: Any) => v.asInstanceOf[Tensor[_]].mapValues(cc)
+                  (v: Any) => {
+                    v match {
+                      case _: SparkVector => sparkVectorToMLeapTensor(v.asInstanceOf[SparkVector]).mapValues(cc)
+                      case _ => v.asInstanceOf[Tensor[_]].mapValues(cc)
+                    }
+                  }
                 case BasicType.String =>
                   val cc = c.asInstanceOf[(Any) => String]
-                  (v: Any) => v.asInstanceOf[Tensor[_]].mapValues(cc)
+                  (v: Any) => {
+                    v match {
+                      case _: SparkVector => sparkVectorToMLeapTensor(v.asInstanceOf[SparkVector]).mapValues(cc)
+                      case _ => v.asInstanceOf[Tensor[_]].mapValues(cc)
+                    }
+                  }
                 case BasicType.ByteString =>
                   val cc = c.asInstanceOf[(Any) => ByteString]
-                  (v: Any) => v.asInstanceOf[Tensor[_]].mapValues(cc)
+                  (v: Any) => {
+                    v match {
+                      case _: SparkVector => sparkVectorToMLeapTensor(v.asInstanceOf[SparkVector]).mapValues(cc)
+                      case _ => v.asInstanceOf[Tensor[_]].mapValues(cc)
+                    }
+                  }
+              }
+          }
+        }.orElse {
+          Some {
+            Try {
+              (v: Any) => {
+                v match {
+                  case _: SparkVector => sparkVectorToMLeapTensor(v.asInstanceOf[SparkVector])
+                  case _ => v
```

Comment:
This is a bit of a deviation from all the other casting logic.

Normally this module returns `None` when `from.base == to.base`, but in this case we must return a function that can cast a Vector into a Tensor